### PR TITLE
MULE-18623: DynamicConfigurationProvider leaks configuration instances on cache eviction

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java [hunk 1]
```diff
@@ -217,7 +220,6 @@ protected void registerConfiguration(ConfigurationInstance configuration) {
   public List<ConfigurationInstance> getExpired() {
     return cache.asMap().entrySet().stream().filter(entry -> isExpired(entry.getValue())).map(entry -> {
       cache.invalidate(entry.getKey());
-      unRegisterConfiguration(entry.getValue());
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java:220]: why is this the problem? Who else can remove items from the cache?

Comment (eze210) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java:220]: There is a list in the parent class which isn't cleaned up

Comment (eze210) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java:220]: Sorry, the cache's removalListener

Comment (marianogonzalez) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java:220]: that's not what I asked. Your fix is to move this from here to the remove listener. My question is, why does that fix the problem? that makes sense if the item gets removed from the cache from any place other than this. Which place is that?

Comment (eze210) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java:220]: The cache has a time-based eviction policy (see `expireAfterAccess(expirationPolicy.getMaxIdleTime(), expirationPolicy.getTimeUnit())` )
It's automatically called on cache maintenance

**Comment by eze210** [comment]
--validate

**Pull request merged**